components/automate-deployment/pkg/server/automate_config.go

```diff
@@ -280,6 +282,46 @@ func setConfigForRedirectLogs(req *api.PatchAutomateConfigRequest, existingCopy
 		}
 	}
 
+	//Set the config if already not set
+	if req.GetConfig().GetGlobal().GetV1().GetLog().GetRedirectSysLog().GetValue() == true &&
+		existingCopy.GetGlobal().GetV1().GetLog().GetRedirectSysLog().GetValue() == true {
+
+		res, err := UpdateByMergingStructs(req, existingCopy)
+		if err != nil {
+			logrus.Errorf("cannot merge requested and existing structs through mergo.Merge: %v", err)
+		}
+
+		if reflect.DeepEqual(res.Config.Global.V1.Log, existingCopy.Global.V1.Log) {
```

Comment:
I suggest we should not use deepEqual directly, since the type conversions in deepEqual are not reliable, instead, we have to manually compare the required fields in both structs.